### PR TITLE
APIMGMT-637: Add info about versioning / lifecycle markings

### PR DESCRIPTION
**Pull request by bszwarc**
# Description

Added info about versioning / lifecycle markings:
beta, latest version, deprecated.

Improved the versioning doc overall

Fixes APIMGMT-637

Staging: https://staging.developer.box.com/guides/api-calls/api-versioning-strategy/#endpoint-marking

### CODE REVIEW

**Review by Percival33 — APPROVED** [review]

File: content/guides/api-calls/api-versioning-strategy.md [hunk 1]
```diff
@@ -69,10 +71,20 @@ We strongly recommend updating your apps to make requests to the latest stable A
...
 version-aware, you anchor it to a specific set of features, ensuring consistent behavior throughout the supported 
 timeframe.
 
+### Endpoint marking
+
+To keep you informed about the current API state, and improve the readability of the versioned API reference, the affected endpoints are marked with the following terms: 
+
+* **Beta**: Endpoints marked with **beta**, are offered subject to Box’s Main Beta Agreement, meaning the available capabilities may change at any time. Although beta endpoints not the same as versioned endpoints, they are a part of API lifecycle.
+
+* **Latest version**: The latest version applies to APIs that are already versioned. **Latest version** marks the most recent API version of an endpoint.
```
Comment (wsocha) [inline content/guides/api-calls/api-versioning-strategy.md:84]: What about adding 'stable' here? `...marks the most recent stable API version...`

File: content/guides/api-calls/api-versioning-strategy.md [hunk 2]
```diff
@@ -52,15 +54,15 @@ Box can introduce a new breaking change to certain endpoints **once per year**.
...
 ### Naming convention
 
 New API versions are labeled according to the calendar year of their release.
-For example, if a new version of the Sign Requests endpoint is released in 2025, it will be named `2025.0`.
 
-Box can issue a new breaking change to API endpoints once per year, reserving the right to release an additional breaking change to address security or privacy concerns. In such cases, the new version will be incremented by one in the suffix.
-For example, if security issues need addressing in the previously released version `2025.0` of Sign Requests, the new version will be labeled `2025.1`.
+**Example**: If a new version of the Sign Requests endpoint is released in 2025, it will be named `2025.0`.
+
+Box can issue a new breaking change to API endpoints **once** per year, reserving the right to release an additional breaking change to address security or privacy concerns. In such cases, the new version will be incremented by one in the suffix.
```
Comment (wsocha) [inline content/guides/api-calls/api-versioning-strategy.md:63]: `...a new major change...`
`...additional major change...`

File: content/guides/api-calls/api-versioning-strategy.md [hunk 3]
```diff
@@ -25,24 +25,26 @@ Box API supports versioning in URL `path` and `header`. To determine which versi
...
-
-If there is a significant change to API behavior, the new endpoint will be exposed under the new URL.
-For example, `https://upload.box.com/api/2.0/files/content` supports a multipart content type when uploading files to Box. If the new version of this API stops supporting this content type, it will be released under a new URL `https://upload.box.com/api/3.0/files/content`.
+If the provided version is correct, a response is sent back to the client. The response also contains the `box-version` header if it was provided in the request. By default, this header is not present in the response. If the version is wrong, an error with the HTTP code `400` is returned to the client.
 
 ## Release schedule and naming convention
 
-Box can introduce a new breaking change to certain endpoints **once per year**. An endpoint encompasses all supported HTTP methods within a specific API root. For example, the Sign Request endpoints include:
+Box can introduce a new breaking change to certain endpoints **once per year**.
```
[the remark this answers is not in the document]
Comment (bszwarc) [reply content/guides/api-calls/api-versioning-strategy.md:44]: Agreed to leave "breaking"

File: content/guides/api-calls/api-versioning-strategy.md [hunk 4]
```diff
@@ -69,10 +71,20 @@ We strongly recommend updating your apps to make requests to the latest stable A
...
 If your request doesn't include a version, the API defaults to the initial Box API version—the version available before 
 year-based versioning was introduced. However, relying on this behavior is not recommended when adopting deprecated 
-changes. To ensure consistency, always specify the API version with each request. By making your application 
+changes. To ensure consistency, always specify the API version, with each request. By making your application 
 version-aware, you anchor it to a specific set of features, ensuring consistent behavior throughout the supported 
 timeframe.
 
+### Endpoint marking
+
+To keep you informed about the current API state, and improve the readability of the versioned API reference, the affected endpoints are marked with the following terms:
```
[the remark this answers is not in the document]
Comment (bszwarc) [reply content/guides/api-calls/api-versioning-strategy.md:80]: Good point, I will add a sentence explaining that the plli actually reflects the tag and is displayed in the UI

**Review by antusus — APPROVED** [review]
👍

**Pull request merged**